src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs

```diff
@@ -496,6 +496,28 @@ public virtual CultureInfo Parent
                     CultureInfo culture;
                     string parentName = _cultureData.ParentName;
 
+                    if (parentName == "zh" && _name.Length == 5 && _name[2] == '-')
```

Comment:
This should only be done when running on ICU, right?

Reply:
I am doing here for both to ensure if Windows changed their behavior we'll still continue to work. when running this check on NLS today, this check will fail and everything will continue work as it is.

Reply:
Ok, makes sense.